mm2src/mm2_main/src/lp_swap/taker_swap_v2.rs

```diff
@@ -456,13 +456,18 @@ impl<MakerCoin: MmCoin + MakerCoinSwapOpsV2, TakerCoin: MmCoin + TakerCoinSwapOp
     }
 
     fn dex_fee(&self) -> DexFee {
-        let taker_pub = self.taker_coin.taker_pubkey_bytes(); // for dex fee calculation we need only permanent (non-derived for HTLC) taker pubkey here
-        DexFee::new_from_taker_coin(
-            &self.taker_coin,
-            self.maker_coin.ticker(),
-            &self.taker_volume,
-            taker_pub.as_deref(),
-        )
+        if let Some(taker_pub) = self.taker_coin.taker_pubkey_bytes() {
```

Comment:
I don't understand `taker_pubkey_bytes` returning None for privacy coins then adding the check here https://github.com/KomodoPlatform/komodo-defi-framework/blob/356eaadbe085a2c67b6ec62c341aabc7daea224a/mm2src/coins/lp_coins.rs#L3804
I think `taker_pubkey_bytes` can be removed.

Reply:
Same answer as this https://github.com/KomodoPlatform/komodo-defi-framework/pull/2112#discussion_r2004202500 I will leave it up to you